### ISSUE TITLE
Inlay hints `auto` type for a lambda is `type`

### ISSUE THREAD

**Issue by Zingam** [microsoft/vscode-cpptools#9971]
Type: <b>Bug</b>

The inlay hint displayed for `auto` for the local variables is `type`:

```c++
  auto even = [](int i) { return 0 == i % 2; };
  auto square = [](int i) { return i * i; };
```

Extension version: 1.13.2
VS Code version: Code 1.72.0 (64bbfbf67ada9953918d72e1df2f4d8e537d340e, 2022-10-04T23:20:47.539Z)
OS version: Linux x64 5.15.0-48-generic snap
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 3400)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off|
|Load (avg)|6, 4, 3|
|Memory (System)|7.42GB (3.52GB free)|
|Process Argv|--no-sandbox --force-user-env --unity-launch --crash-reporter-id 20280fc3-3c6d-43d3-9eb5-55c536b003d8|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu-wayland|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu-wayland|
|XDG_SESSION_TYPE|wayland|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vscaat:30438848
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
2144e591:30553903
vsccc:30566497
pyindex848cf:30577861
40g7c324:30573242

```

</details>

![image](https://user-images.githubusercontent.com/3337419/194612353-06ff6c28-7a81-463b-b16b-aad2f30fdc86.png)

**Comment by sean-mcmanus** [comment]
I don't repro the issue -- can you provide more repro info? What compiler and IntelliSense mode are you using, cppStandard, etc. What inlayHint settings are you using?

![image](https://user-images.githubusercontent.com/19859882/194616134-dbdadfff-77a3-48bf-af57-1c9c3485686d.png)

**Comment by Zingam** [comment]
It's the same garbage sample project I've been using to report other issues:
[Test.zip](https://github.com/microsoft/vscode-cpptools/files/9736001/Test.zip)

The OS is Ubuntu 22.04 with the GCC 12.2 from the distribution and Clang from here https://apt.llvm.org/
I am switching between VSCode Clang Debug and VSCode GCC debug profiles and rebuilding. 

I have enabled all inlay hints but *Show Space*.

**Comment by Zingam** [comment]
I suppose I am using the default mode:

![image](https://user-images.githubusercontent.com/3337419/194618264-968be5d4-f93b-446b-9d87-14558766571a.png)

**Comment by Zingam** [comment]
![image](https://user-images.githubusercontent.com/3337419/194618390-5824781d-c96f-446a-9754-2878f3c2ade1.png)

**Comment by sean-mcmanus** [comment]
Yeah, I reproed it now -- with clang 15. With older gcc/clang it doesn't repro.

**Comment by sean-mcmanus** [comment]
Actually, it looks like it starts reproing after "#include <ranges>" is added (even with gcc 11), which seems very strange (i.e. like a bug).

**Comment by sean-mcmanus** [comment]
Starts to repro when the lambda is passed to std::views::filter.

```cpp
#include <ranges>

int main()
{
    auto even = [](int i)
    { return 0 == i % 2; };
    auto square = [](int i)
    { return i * i; };
    std::views::filter(even);
    return 0;
}
```

![image](https://user-images.githubusercontent.com/19859882/194621471-973e0c07-2864-4121-8185-0d6982b300fb.png)

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3
